R/wwinference.R

```diff
@@ -41,31 +41,102 @@
 #' @param compiled_model The pre-compiled model as defined using
 #' `compile_model()`
 #'
-#' @return A nested list of the following items, intended to allow the user to
-#' quickly and easily plot results from their inference, while also being able
-#' to have the full user functionality of running the model themselves in stan
-#' by providing the raw model object and diagnostics. If the model runs, this
+#' @return An object of the `ww_inference_fit` class containing the following
+#' items that are intended to be passed to downstream functions to do things
+#' like extract posterior draws, get diangostic behavior, and plot results
+#' (for example). If the model runs, this
 #' function will return:
-#' `draws_df`: A tibble containing the full set of posterior draws of the
-#' estimated, nowcasted, and forecasted: counts, site-level wastewater
-#' concentrations, "global"(e.g. state) R(t) estimate, and the  "local" (site +
-#' the one auxiliary subpopulation) R(t) estimates. In the instance where there
-#' are observations, the data will be joined to each draw of the predicted
-#' observation to facilitate plotting.
-#' `raw_fit_obj`: The CmdStan object that is returned from the call to
+#' `fit`: The CmdStan object that is returned from the call to
 #' `cmdstanr::sample()`. Can be used to access draws, summary, diagnostics, etc.
-#' `date_time_spine`: Mapping from time in stan to dates
-#' `lab_site_spine`: Mapping from lab_site_index in stan to lab and site
-#' `subpop_spine`: Mapping from site index in stan to site
+#' `input_data`: a list containing the input `ww_data` and the input
+#' `count_data` used in the model.
+#' `stan_data_args`: a list containing the inputs passed directly to the
```

Comment:
it feels like the names here are inconsistent. 

I think mcmc_options should be fit_opts or similar as this allows you to generalise if you don't use mcmc. I usually go with data_llist for stan_data_args but personal preference

Reply:
Can change to `fit_opts` that makes sense. 
I have slight preference for `stan_data_args` bc we return an `input_data` list (`ww_data`, `hosp_data`, and eventually `dist_matrix` which might be confused with `data_list`).

Reply:
I think I might call the input data `raw_` but I don't feel strongly

Reply:
So `input_data` -> `fit_opts` and `stan_data_args` stays?

Reply:
I'd propose `stan_data_list` instead of `stan_data_args`

Reply:
Hmm no I think we want (and @gvegayon I will do this as a PR into this branch):
`mcmc_options` -> `fit_opts`
`stan_data_args` -> `stan_data_list`
`input_data` -> `raw_input-data`